### ISSUE TITLE
Wan2.2 TAEHV training detail

### ISSUE THREAD

**Issue by Leoooo333** [madebyollin/taehv#17]
Hi there, it's so amazing to see the almost 60x acceleration on Wan2.2 VAE (16x16x4) one. Thanks so much for the great work. I currently want to reimplement the MotionStream paper, they mention their VAE follow taehv, but they don't open-source yet.  So I want to ask if you are pleased to share the details for the Wan2.2 taehv training, like how many hours videos, at what resolution, and how many iterations or estimated hours to get a good enough weights? 

I have launch several training from scratch following seraena, but seems they all converged at around SSIM=0.89 after 30k iterations (batchsize=64), while your taehv can achieve SSIM=0.94.

**Comment by shijx12** [comment]
Hi @madebyollin , thanks for your great work! I noticed that when using TAEW2.2 on a larger resolution like (480, 832), there are dark ripples or streaks in the reconstructed video. So I am trying to finetune the weights with more data. 

1. Finetuning the decoder is ok, with L1 reconstruction loss + GAN loss.
2. Finetuning the encoder doesn't work. By aligning tae.encode(x) to vae.encode(x).mode() or vae.encode(x).sample() with mse loss or L1 loss, it will always cause the reconstructed video to become blurry, and reduce the motion coherence across four consecutive frames. Have you encountered similar issues during training? Do you have any troubleshooting ideas or suggestions? Thank you very much!

**Comment by madebyollin** [comment]
@shijx12 Oh cool! Finetuning TAEW2.2 on larger resolution sounds like a great idea - I have so far just been training at fairly low resolution on one GPU and hoping it generalizes 😅 which isn't super reliable.

To reduce encoder problems, I have used a feature-matching loss (see also [here](https://github.com/madebyollin/taesd/issues/11#issuecomment-3629582150)) to force the encoder to generate "plausible" encodings even when the tiny encoder can't match the reference encoder exactly.

<img width="609" height="96" alt="Image" src="https://github.com/user-attachments/assets/2811bb54-58cc-4752-8a29-3fe43c4cc60d" />

The features I've used come from the averaged (no-gradient) decoder before the first upsample layer:

<img width="431" height="89" alt="Image" src="https://github.com/user-attachments/assets/d6c3a23a-126b-45cc-ba2f-55a90a7e631d" />

You could go even further and use GAN loss on the encoder too, or match more features from the small decoder, or even match features from the full-size decoder too, but matching small-decoder pre-upsample features should be a good place to start.

**Comment by shijx12** [comment]
@madebyollin Thanks for your suggestions! I have tried the feature-matching loss, together with L1 loss in the latent space, and LPIPS loss + gan loss on the reconstructed video. 
However, my finetuned tae encoder + my finetuned tae decoder still output blurry results (like Col 3 in the example, where Col 1 is the raw video), while original vae encoder + my finetuned tae decoder is better (Col 2 in the example). It seems that the tae encoder is hard to capture the detailed textures and motions. Col 4 is the reconstructed result with your original tae, the details are much better than my finetuned version. So I am curious that do you have other training tricks or important hyper-parameters to achieve this performance?
Besides, I noticed that reconstructed video with your original tae (Col 4) exhibits a frame-level temporal delay of several frames relative to the raw video. I guess maybe it is due to the padding methods? To pad the video to 4n+1 frames, repeated frames are appended to the end. While in vae implementation, padding is happened in the start.

https://github.com/user-attachments/assets/a60791d9-7a8e-465e-bf4b-bb37a3effa15

**Comment by madebyollin** [comment]
@shijx12 One other trick to mention is that I train the TAE decoder on *both* real-encoder latents and TAE-encoder latents (7/8 real 1/8 TAE typically) to reduce compounding error when using TAEHV enc+dec. However, I detach latents before decoding (to not backprop the decoder losses into the encoder), so this does not explain your "official TAE encoder + my decoder is sharper than my TAE encoder + my decoder" results. A time alignment bug (or just undertraining / not having enough featurematching weight?) would seem more likely.

Regarding time alignment, though, TAEW2.2's temporal behavior matches the official VAE in my testing (see [this test notebook](https://colab.research.google.com/drive/1a4aog4h0hYRnC9sm15NgaaJu_Tb8maxX?usp=sharing)), so I'm not sure where the delay you're seeing comes from.

<img width="1354" height="664" alt="Image" src="https://github.com/user-attachments/assets/f1df22af-508a-47ed-80cd-d7994f5e92fc" />

If you possible, please isolate the conditions for reproducing the delay down to a small modification on top of that test notebook, and then I can investigate further.

**Comment by iamweiweishi** [comment]
Hi,
I tried to train one small WAN2.2-VAE, but I find that the results look very noisy and lots of aging skin texture appeared on human faces.
May I know how to solve the issues.
Thank you. 

<img width="1173" height="987" alt="Image" src="https://github.com/user-attachments/assets/2bdf26bf-8655-450a-b93c-b79c221c6bc0" />

**Comment by madebyollin** [comment]
> Hi, I tried to train one small WAN2.2-VAE, but I find that the results look very noisy and lots of aging skin texture appeared on human faces. May I know how to solve the issues. Thank you.

Hmm, I didn't have any tests or fixes targeted at improving human faces specifically. But the loss terms responsible for controlling this kind of generated detail are **1)** feature matching loss (I typically use LPIPS, but you can use DISTS, DINO, whatever) and **2)** GAN loss (either standalone or conditioned on the encoded latent - I have typically used a PatchGAN style latent-conditioned discriminator on 4-6 frame windows). If the generated detail doesn't seem sharp or realistic enough to meet your needs, I would recommend improving the strength/capacity/sophistication of the feature-matching and adversarial losses. You could reference e.g. https://arxiv.org/abs/2607.14898v1 for ideas

**Comment by iamweiweishi** [comment]
> > Hi, I tried to train one small WAN2.2-VAE, but I find that the results look very noisy and lots of aging skin texture appeared on human faces. May I know how to solve the issues. Thank you.
> 
> Hmm, I didn't have any tests or fixes targeted at improving human faces specifically. But the loss terms responsible for controlling this kind of generated detail are **1)** feature matching loss (I typically use LPIPS, but you can use DISTS, DINO, whatever) and **2)** GAN loss (either standalone or conditioned on the encoded latent - I have typically used a PatchGAN style latent-conditioned discriminator on 4-6 frame windows). If the generated detail doesn't seem sharp or realistic enough to meet your needs, I would recommend improving the strength/capacity/sophistication of the feature-matching and adversarial losses. You could reference e.g. https://arxiv.org/abs/2607.14898v1 for ideas

OK.

Thank you. I will have a try.
